### PR TITLE
Milestone 3 Game logic

### PR DESCRIPTION
**Pull request by tongoonamujera**
this milestone consists of real game logic

### CODE REVIEW

File: lib/game_player_helper.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    end
+  end
+end
+
+include Player_helper
+player = Player_checks.new
+puts "hey Whats your name: "
+name = gets.chomp
+puts player.check_name_validity?(name)
+
```
Comment (george-shammar) [inline lib/game_player_helper.rb:41]: - [ ] No puts/print and gets.chomp outside main.rb file.

**Review by george-shammar — CHANGES_REQUESTED** [review]
### **STATUS:CHANGES REQUIRED** :recycle: :recycle: :construction: :construction_worker_man: 

Good Job tongoona. 

**Highlights**
Your repo looks professional with the correct workflow and README. :+1: 

Please see the required changes below as well as my inline comments.

- [ ] No puts/print and gets.chomp outside main.rb file. Your main.rb should be responsible for that. See my in-line comment in this regard.
- [ ] Your TicTacToe game crashes after asking for first player's name. The output is a "Loaderror". Cannot load "/game_board.rb". The file does not exist but you have required it in your code.

Since I was unable to check the behaviour of your game, the requirement for the behaviour of your game would be checked upon your next submission. 

Please submit another CRR when you're ready.

Cheers and happy coding. :smile:

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: :stop_sign: 
----------------
Hello @tongoonamujera, great job working on the required changes :muscle: :1st_place_medal: 
However, there are a few changes to make to complete this milestone :hammer_and_wrench: see them below :point_down: 
---------------------

- [ ] You have some validation for the names but the games continue if a keep input the same value. 
Screenshot from 2021-06-20 20-48-00

Please fix this bug. 

- [ ] If a make a move that is already taken, on the next move put the same mark for the move.

Screenshot from 2021-06-20 20-51-06

- [ ] The game does not change the mark for the next player's move. 

Screenshot from 2021-06-20 20-52-42

-----------------
Please fix these required changes and send it again for another review :muscle: 
HAPPY CODING!!!!

**Review by emmanuellekamwa — CHANGES_REQUESTED** [review]
**STATUS: CHANGES REQUESTED 🛑🛠**
Hi @tongoonamujera 👋🏽
You have done a great job so far but there are still some changes that must be done to have this milestone approved. Check it below 👇🏽

- [ ] As the previous TSE pointed out, the token of the next player does not change after his move. Kindly fix that.
   That's all. Just apply that change and request for another code review. Good luck and happy coding 👨🏽‍💻👨🏽‍💻.

**Review by bini-i — APPROVED** [review]
# Status: Approved :heavy_check_mark:          
### Hello @tongoonamujera   :wave: 

Thanks for working on the requirements by the previous reviewers, your project meets all the requirements and it is approved :tada: .
#### Highlights
- Methods are small and do one thing only, which makes your code easy to read and debug.
- Only one method(**winner?**) is used to check for winning cases, which makes your code DRY.
- Note: Make sure to look out for attr_accessor as it allows setting instance properties from outside the class, only use them when needed.

### Keep up the good work :+1: 

#### If you have any question feel free to ask within the PR and ping me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (Code Reviewer)
> GitHub: [@bini-i](https://github.com/bini-i) 
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)#

**Pull request merged**